Website/Providers/DataProviders/SqlDataProvider/DotNetNuke.Schema.SqlDataProvider

```diff
@@ -17246,9 +17251,23 @@ GO
 PRINT N'Creating {databaseOwner}[{objectQualifier}GetPasswordHistory]'
 GO
 CREATE PROCEDURE {databaseOwner}[{objectQualifier}GetPasswordHistory]
-    @UserID			int
-AS
-        SELECT * from {databaseOwner}{objectQualifier}PasswordHistory where UserID=@UserID
+	@UserID             int,
+	@PasswordsRetained  int,
+	@DaysRetained       int
+AS
+	DELETE FROM {databaseOwner}{objectQualifier}PasswordHistory 
+	WHERE UserID=@UserId 
+	  AND PasswordHistoryID NOT IN (
+		SELECT TOP (@PasswordsRetained) PasswordHistoryID 
+		FROM {databaseOwner}{objectQualifier}PasswordHistory
+		WHERE UserID=@UserId 
+		ORDER BY CreatedOnDate DESC
+		)
+	  AND DATEDIFF(day, CreatedOnDate, GETDATE()) > @DaysRetained;
+
+	SELECT * 
+	FROM {databaseOwner}{objectQualifier}PasswordHistory 
+	WHERE UserID=@UserID
 GO
 PRINT N'Creating {databaseOwner}[{objectQualifier}SetFileHasBeenPublished]'
 GO
```

Comment:
You shouldn't change anything in this file as the individual migrations will take care of this.